src/database-layer/db.ts

```diff
@@ -710,6 +717,27 @@ if (maybePg != null) {
 		return {
 			engine: Engines.postgres,
 			executeSql: atomicExecuteSql,
+			on: async (name, fn, options) => {
+				if (name !== 'notification') {
+					throw new Error(`Unsupported listener type: ${name}`);
+				} else if (options?.channel == null) {
+					throw new Error('Missing channel option for notification listener');
+				} else if (options.channel.includes('"')) {
+					throw new Error(
+						`Invalid channel name for task LISTEN: ${options.channel}`,
+					);
+				}
+
+				const client = await pool.connect();
```

Comment:
Have you tested what happens if the connection drops? eg due to a db restart/failover?

Reply:
It seems that the client is released and a reconnect is not attempted. I'll look at adding some bits to attempt automatic reconnects.